src/content/docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/configure-control-plane-monitoring.mdx

```diff
@@ -0,0 +1,499 @@
+* [OpenShift](http://learn.openshift.com/?extIdCarryOver=true&sc_cid=701f2000001OH7iAAG) 4.x uses control plane component metric endpoints that are different than the default.
+
+## Control plane component [#component]
+
+The task of monitoring the Kubernetes control plane is a responsibility of the `nrk8s-controlplane` component, which by default is deployed as a DaemonSet. This component is automatically deployed to master nodes, through the use of a default list of `nodeSelectorTerms` which includes labels commonly used to identify master nodes, such as `node-role.kubernetes.io/control-plane` or `node-role.kubernetes.io/master`. Regardless, this selector is are exposed in the `values.yml` file and therefore can be reconfigured to fit other environments.
+
+Clusters that do not have any node matching these selectors will not get any pod scheduled, thus not wasting any resources and being functionally equivalent of disabling control plane monitoring altogether by setting `controlPlane.enabled` to `false` in the Helm Chart.
+
+Each component of the control plane has a dedicated section, which allows to individually:
+
+* Enable or disable monitoring of that component
+* Define specific selectors and namespaces for discovering that component
+* Define the endpoints and paths that will be used to fetch metrics for that component
+* Define the authentication mechanisms that need to be used to get metrics for that component
+* Manually specify endpoints that skip autodiscovery completely[^staticEndpoints]
+
+[^staticEndpoints]: Configuring static endpoints requires careful configuration, please make sure you completely understand all the caveats mentioned in the [static endpoints section](#staticEndpoints).
+
+### Autodiscovery and default configuration [#autodiscovery-default]
+
+By default, our [Helm Chart](/docs/kubernetes-pixie/kubernetes-integration/installation/install-kubernetes-integration-using-helm) ships a default configuration that should work out of the box for some control plane components for on-premise distributions that run the control plane inside of the cluster, such as `Kubeadm` or `minikube`.
+
+#### `hostNetwork` and `privileged` [#hostnetwork-privileged]
+
+Most users and Kubernetes distributions configure the control plane metrics endpoints to listen only in the loopback interface, i.e. `localhost`. For this reason, the control plane component is deployed with `hostNetwork: true` by default when `privileged` is set to `true` (the default).
+
+To honor the master `privileged` setting, `hostNetwork` will be set to `false` if `privileged` is set to `false`. This will cause control plane scraping to fail in most environments, which will result in missing metrics or the `nrk8s-controlplane` pods getting stuck into a `CrashLoopBackoff` state. This is a limitation of Kubernetes itself, as control plane cannot be monitored without `hostNetwork` unless components are manually configured to do so.
```

Comment:
I find this sentence a bit confusing 
"To honor the master `privileged` setting, `hostNetwork` will be set to `false` if `privileged` is set to `false`."

Reply:
Should we add a working in the chart to warning explicitly CrashLoopBackoff? Or is that a note next to privileged options in values?

Reply:
I reworded it a bit, let me know what you think.

Regarding the `CrashLoopBackoff`, it would depend on the user's env. It might fail if using `localhost` but only one node has the component on it, or it might just duplicate metrics silently if the url is non-local.

Reply:
resolved